zotero/src/data_structure/item/mod.rs

```diff
@@ -204,17 +497,87 @@ pub struct Creator {
     pub last_name: String,
 }
 
+impl Creator {
+    pub fn full_name(&self) -> String {
+        format!("{} {}", self.first_name, self.last_name)
+    }
+
+    pub fn short_name(&self) -> String {
+        if self.first_name.len() == 0 {
+            return format!("{}", self.last_name)
+        }
+        format!("{}. {}", self.first_name.chars().next().unwrap(), self.last_name)
```

Comment:
```suggestion
match self.first_name.chars().next() {
    Some(first_initial) => format !("{}. {}", first_initial, self.last_name),
    None => format !("{}", self.last_name),
}
```
This code doesn't work for some names: `Jean-Philippe` should be transformed into `J.-P.` or `Hanna Mary` into `H. M.`

Perhaps we should declare the surname as optional and set the value to None when deserializing an empty string?

Reply:
i changed the code to the suggest logic, a more though update would be needed to enable smarter handling of names, we should probably write some test cases to to ensure that it is handled correctly